File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @rna_transcriptions %{
    ?A => ?U,
    ?C => ?G,
    ?G => ?C,
    ?T => ?A
  }

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RNATranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna), do: Enum.map(dna, &Map.fetch!(@rna_transcriptions, &1))
end

File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

-define(RNA_TRANSCRIPTIONS, #{$G => $C, $C => $G, $T => $A, $A => $U}).


to_rna(Strand) ->
  lists:map(fun fetch_rna/1, Strand).

% Private

fetch_rna(Dna) ->
  maps:get(Dna, ?RNA_TRANSCRIPTIONS).
